client/my-sites/customer-home/cards/features/domain-upsell/index.jsx

```diff
@@ -13,15 +13,17 @@ import { domainRegistration } from 'calypso/lib/cart-values/cart-items';
 import CalypsoShoppingCartProvider from 'calypso/my-sites/checkout/calypso-shopping-cart-provider';
 import useCartKey from 'calypso/my-sites/checkout/use-cart-key';
 import { isCurrentUserEmailVerified } from 'calypso/state/current-user/selectors';
+import { getDomainsBySiteId } from 'calypso/state/sites/domains/selectors';
 import { getSelectedSite, getSelectedSiteSlug } from 'calypso/state/ui/selectors';
 
 import './style.scss';
 
 export default function DomainUpsell() {
 	const site = useSelector( ( state ) => getSelectedSite( state ) );
 	const isEmailVerified = useSelector( ( state ) => isCurrentUserEmailVerified( state ) );
+	const siteDomains = useSelector( ( state ) => getDomainsBySiteId( state, site.ID ) );
 
-	if ( ! isEmailVerified || ! isFreePlanProduct( site.plan ) ) {
+	if ( siteDomains.length > 1 || ! isEmailVerified || ! isFreePlanProduct( site.plan ) ) {
```

Comment:
When we have a registered domain, even on a free plan, we got two domains the `.wordpress.com` and the registered domain itself. So this is working great!

Reply:
Thanks for testing this!